hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java

```diff
@@ -0,0 +1,357 @@
+    this.catalogName = catalogName;
+    this.metaClients = metaClients;
+    this.fileIO = fileIO;
+    this.fullName = CatalogUtil.fullTableName(catalogName, viewIdentifier);
+    this.database = dbName;
+    this.viewName = viewIdentifier.name();
+    this.maxHiveTablePropertySize =
+        conf.getLong(HIVE_TABLE_PROPERTY_MAX_SIZE, HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT);
+  }
+
+  @Override
+  public void doRefresh() {
+    String metadataLocation = null;
+    Table table;
+
+    try {
+      table = metaClients.run(client -> client.getTable(database, viewName));
+      HiveOperationsBase.validateTableIsIcebergView(table, fullName);
+
+      metadataLocation =
+          table.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+
+    } catch (NoSuchObjectException e) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s.%s", database, viewName);
+      }
+    } catch (TException e) {
+      String errMsg =
+          String.format("Failed to get view info from metastore %s.%s", database, viewName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  @Override
+  public void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    boolean newView = base == null;
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+    boolean hiveEngineEnabled = false;
+
+    CommitStatus commitStatus = CommitStatus.FAILURE;
+    boolean updateHiveView = false;
+
+    HiveLock lock = lockObject();
+    try {
+      lock.lock();
+
+      Table tbl = loadHmsTable();
+
+      if (tbl != null) {
+        // If we try to create the view but the metadata location is already set, then we had a
+        // concurrent commit
+        if (newView
+            && tbl.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP)
+                != null) {
+          throw new AlreadyExistsException(
+              "%s already exists: %s.%s",
+              tbl.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())
+                  ? ContentType.VIEW.value()
+                  : ContentType.TABLE.value(),
+              database,
+              viewName);
+        }
+
+        updateHiveView = true;
+        LOG.debug("Committing existing view: {}", fullName);
+      } else {
+        tbl =
+            newHmsTable(
+                PropertyUtil.propertyAsString(
+                    metadata.properties(),
+                    HiveCatalog.HMS_TABLE_OWNER,
+                    HiveHadoopUtil.currentUser()));
+        LOG.debug("Committing new view: {}", fullName);
+      }
+
+      tbl.setSd(
+          HiveOperationsBase.storageDescriptor(
+              metadata.schema(),
```

Comment:
HMS Table resource has 2 fields (ViewOriginalText and ViewExpandedText) for storing view query. What is the reason for not storing the view query in those fields ?

Reply:
Thanks @vteja11 for reviewing. I have added it back.